lib/multipass_ex/crypt.ex

```diff
@@ -0,0 +1,116 @@
+defmodule MultipassEx.Crypt do
+  @moduledoc """
+  Module for encrypting and decrypting data in compliance with the Ruby MultiPass
+  gem.
+
+  AES_CBC128 is the encryption algorithm used.
+
+  __Warning__: The ruby gem uses an insecure IV and it's mirrored here to be compatible
+  since it is not sent with the Multipass data.
+  """
+  require Logger
+
+  alias MultipassEx.{DecodingError, EncodingError, CryptError}
+  alias MultipassEx.Coding
+
+  @algorithm :aes_cbc128
+  @block_size 16
+  @key_byte_length 16
+  @iv String.duplicate(<<0>>, 16)
+
+  @spec encrypt(map(), binary(), binary()) ::
+          {:ok, binary()}
+          | {:error, EncodingError.t()}
+  def encrypt(data, site_key, api_key) do
+    key = generate_key(site_key, api_key)
+
+    prepared_data = prepare_data_for_encryption(data)
+
+    encrypted_data =
+      @algorithm
+      |> :crypto.block_encrypt(key, @iv, prepared_data)
+      |> Coding.encode()
+
+    {:ok, encrypted_data}
+  rescue
+    Jason.EncodeError ->
```

Comment:
so you've got your data preparer neatly abstracted out into its own function `prepare_data_for_encryption/1`, but you're leaking your preparation method by rescuing Jason errors here.

you might instead consider that your preparation function return either `{:ok, prepared_data}` or `{:error, %EncodingError{}}`

Reply:
good point 👍